drivers/nutdrv_qx_masterguard.c

```diff
@@ -0,0 +1,1049 @@
+	{ "battery.packs",		ST_FLAG_RW,	masterguard_r_batpacks,	"GS,XX\r",	"",	0,	'(',	"",	7,	7,	"%.0f",	QX_FLAG_SEMI_STATIC | QX_FLAG_RANGE,	masterguard_add_slaveaddr,	NULL,				NULL },
+	{ "battery.packs",		0,		NULL,			"SS--%1d-,XX\r","",	0,	'\0',	"",	0,	0,	"d",	QX_FLAG_SETVAR | QX_FLAG_RANGE,		masterguard_add_slaveaddr,	NULL,				masterguard_setvar },
+/*!! which QX_FLAGs to use?? (changed by instcmd) */
+	{ "ups.beeper.status",		0,		NULL,			"GS,XX\r",	"",	11,	'(',	"",	9,	9,	NULL,	QX_FLAG_SEMI_STATIC,			masterguard_add_slaveaddr,	NULL,				masterguard_beeper_status },
+	/* set with beeper.{en,dis}able */
+
+	/*
+	 * > [GBS,XX\r]
+	 * < [(XX,CCC hhhh HHHH AAAA BBBB DDDD EEE SS.SS\r]
+	 *    0123456789012345678901234567890123456789012
+	 *    0         1         2         3         4
+	 *    (00,100 0017 0000 0708 0712 0994 115 02.28
+	 *    (01,000 0000 0360 0708 0712 0994 076 02.30
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags	precmd				preans	preproc */
+	{ "battery.charge",	0,	NULL,	"GBS,XX\r",	"",	43,	'(',	"",	4,	6,	"%.0f",	0,	masterguard_add_slaveaddr,	NULL,	NULL },
+	/* 
+	 * hhhh: hold time (minutes)
+	 * HHHH: recharge time to 90% (minutes)
+	 * AAAA: Ageing factor (promilles)
+	 * BBBB: Ageing factor time dependant (promilles)
+	 * DDDD: Ageing factor cyclic use (promilles)
+	 * EEE: Calibration factor (percent)
+	 * SS.SS: Actual battery cell voltage
+	*/
+
+	/*
+	 * > [GSN,XX\r]
+	 * < [(XX,SSN=SSSSSSSSSSSnnnnn\r]
+	 *    0123456789012345678901234
+	 *    0         1         2
+	 *    (00,SSN=6A1212      2782
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags				precmd				preans	preproc */
+	{ "device.part",	0,	NULL,	"GSN,XX\r",	"",	25,	'(',	"",	8,	18,	"%s",	QX_FLAG_STATIC | QX_FLAG_TRIM,	masterguard_add_slaveaddr,	NULL,	NULL },
+	{ "device.serial",	0,	NULL,	"GSN,XX\r",	"",	25,	'(',	"",	20,	23,	"%s",	QX_FLAG_STATIC,			masterguard_add_slaveaddr,	NULL,	NULL },
+
+	/*
+	 * > [DRC,XX\r]
+	 * < [(XX,TTTT:hh:mm:ss\r]
+	 *    012345678901234567
+	 *    0         1
+	 *    (00,1869:19:06:37
+	 */
+	/* type			flags		rw	command		answer	len	leading	value	from	to	dfl	qxflags			precmd				preans	preproc */
+	/* this is not really the uptime, but the running time since last maintenance */
+	{ "device.uptime",	ST_FLAG_RW,	NULL,	"DRC,XX\r",	"",	17,	'(',	"",	4,	16,	"%.0f",	QX_FLAG_SEMI_STATIC,	masterguard_add_slaveaddr,	NULL,	masterguard_tttt_hh_mm_ss },
+	{ "device.uptime",	0,		NULL,	"SRC%s,XX\r",	"",	0,	'\0',	"",	0,	0,	"thms",	QX_FLAG_SETVAR,		masterguard_add_slaveaddr,	NULL,	masterguard_setvar },
+
+	/*
+	 * > [MSO\r]
+	 * < [(220 230 240\r]
+	 *    0123456789012
+	 *    0         1
+	 *    (220 230 240
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags				precmd	preans	preproc */
+	{ "output_voltages",	0,	NULL,	"MSO\r",	"",	5,	'(',	"",	1,	0,	"%s",	QX_FLAG_STATIC | QX_FLAG_NONUT,	NULL,	NULL,	masterguard_output_voltages },
```

Comment:
@aquette @clepple : Any suggestions for standard field naming?

Reply:
If this is a set of selectable output voltages, it seems like it would be best to list them as enum options in an R/W variable, but I realize that might be difficult with the qx framework (haven't looked at that much). I have no preference, other than matching an existing name prefix like `output.voltage.*`

Reply:
As I wrote in my original message:

I need to issue a command to the UPS in order to learn the supported values for **output.voltage.nominal**. In the Q* framework, is there a way to simply ignore the result (after it has been digested in a preprocess function to set the corresponding `info_rw_t` enum, of course)? I currently use a custom var **output_voltages** for it.